packages/core/src/telemetry/activity-detector.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Tracks user activity state to determine when memory monitoring should be active
+ */
+export class ActivityDetector {
+  private lastActivityTime: number = Date.now();
+  private readonly idleThresholdMs: number;
+
+  constructor(idleThresholdMs: number = 30000) {
+    this.idleThresholdMs = idleThresholdMs;
+  }
+
+  /**
+   * Record user activity (called by CLI when user types, adds messages, etc.)
+   */
+  recordActivity(): void {
+    this.lastActivityTime = Date.now();
+  }
+
+  /**
+   * Check if user is currently active (activity within idle threshold)
+   */
+  isUserActive(): boolean {
+    const timeSinceActivity = Date.now() - this.lastActivityTime;
+    return timeSinceActivity < this.idleThresholdMs;
+  }
+
+  /**
+   * Get time since last activity in milliseconds
+   */
+  getTimeSinceLastActivity(): number {
+    return Date.now() - this.lastActivityTime;
+  }
+
+  /**
+   * Get last activity timestamp
+   */
+  getLastActivityTime(): number {
+    return this.lastActivityTime;
+  }
+}
+
+// Global activity detector instance
+let globalActivityDetector: ActivityDetector | null = null;
+
+/**
+ * Initialize global activity detector
+ */
+export function initializeActivityDetector(
+  idleThresholdMs: number = 30000,
+): ActivityDetector {
+  if (!globalActivityDetector) {
+    globalActivityDetector = new ActivityDetector(idleThresholdMs);
+  }
+  return globalActivityDetector;
+}
+
+/**
+ * Get global activity detector instance
+ */
+export function getActivityDetector(): ActivityDetector | null {
+  return globalActivityDetector;
+}
+
+/**
+ * Record user activity (convenience function for CLI to call)
+ */
+export function recordUserActivity(): void {
+  // Do not implicitly initialize here to avoid locking in default timeout
```

Comment:
this logic seems a bit more complicated then needed. Do we really need a threshold other than the default? if we don't we can just write
const globalActivityDetector = new ActivityDetector(); 
and keep this a lot simpler. Later if we need more configurability we can always add it.